frontend/packages/kubevirt-plugin/src/k8s/patches/vm/vm-cdrom-patches.ts

```diff
@@ -40,112 +40,113 @@ const assignBootOrderIndex = (vm: VMLikeEntityKind, currDevBootOrder = -1) => {
 
 export const getCDsPatch = async (vm: VMLikeEntityKind, cds: CD[]) => {
   let newBootOrder = assignBootOrderIndex(asVM(vm));
-  let DISKS = getDisks(asVM(vm));
+
+  let DISKS = getDisks(asVM(vm)).filter(
+    (disk) => !disk.cdrom || cds.find((modalCD) => disk.name === modalCD.name || modalCD.newCD),
+  );
   let VOLS = getVolumes(asVM(vm));
   let DATATEMPLATES = getDataVolumeTemplates(asVM(vm));
   const storageClassConfigMap = await getStorageClassConfigMap({ k8sGet });
 
-  cds
```

Comment:
For Reviewers
The only change here is removal of `cds.filter((cd) => cd.changed)`